qa/acceptance-tests/src/test/java/io/camunda/it/cluster/DynamicNodeIdTest.java

```diff
@@ -0,0 +1,142 @@
+                              "camunda.cluster.dynamic-node-id.type",
+                              "s3",
+                              "camunda.cluster.dynamic-node-id.s3.taskId",
+                              taskId + "nodeId=" + app.nodeId(),
+                              "camunda.cluster.dynamic-node-id.s3.bucketName",
+                              bucketName,
+                              "camunda.cluster.dynamic-node-id.s3.leaseDuration",
+                              LEASE_DURATION,
+                              "camunda.cluster.dynamic-node-id.s3.endpoint",
+                              S3.getEndpoint(),
+                              "camunda.cluster.dynamic-node-id.s3.region",
+                              S3.getRegion(),
+                              "camunda.cluster.dynamic-node-id.s3.accessKey",
+                              S3.getAccessKey(),
+                              "camunda.cluster.dynamic-node-id.s3.secretKey",
+                              S3.getSecretKey())))
+          .build();
+
+  @BeforeAll
+  // Setup as static so it's done before the cluster
+  public static void setupAll() {
+    bucketName = UUID.randomUUID().toString();
+
+    taskId = UUID.randomUUID().toString();
+    clusterSize = 3;
+    s3Client =
+        S3NodeIdRepository.buildClient(
+            new S3ClientConfig(
+                Optional.of(new S3ClientConfig.Credentials(S3.getAccessKey(), S3.getSecretKey())),
+                Optional.of(Region.of(S3.getRegion())),
+                Optional.of(S3.getEndpoint())));
+
+    // bucket must be created before the application is started
+    s3Client.createBucket(b -> b.bucket(bucketName));
+  }
+
+  @Test
+  public void shouldStartAppCorrectlyAndAcquireALease() throws IOException {
+    final var objectsInBucket = s3Client.listObjects(b -> b.bucket(bucketName)).contents();
+
+    Awaitility.await("Repository has been initialized")
+        .untilAsserted(() -> assertThat(objectsInBucket).hasSize(clusterSize));
+
+    final var leases = new ArrayList<Lease>();
+    for (final var object : objectsInBucket) {
+      final var lease = s3Client.getObject(b -> b.bucket(bucketName).key(object.key()));
+      final var payload = lease.readAllBytes();
+      if (payload.length > 0) {
+        final var parsed = Lease.fromJsonBytes(OBJECT_MAPPER, payload);
+        if (parsed.isStillValid(System.currentTimeMillis(), LEASE_DURATION)) {
+          leases.add(parsed);
+        }
+      }
+    }
+
+    assertThat(leases).hasSize(3);
+    final var configuredTaskIds =
+        testCluster.brokers().values().stream()
+            .map(
+                app ->
+                    app.property("camunda.cluster.dynamic-node-id.s3.taskId", String.class, null))
+            .collect(Collectors.toSet());
+    assertThat(configuredTaskIds).allSatisfy(o -> assertThat(o).isNotNull());
+    assertThat(leases.stream().map(Lease::taskId).collect(Collectors.toSet()))
+        .isEqualTo(configuredTaskIds);
```

Comment:
:x: Would it be possible to verify that the broker id is assigned correctly from the leased node id?

Reply:
We should also verify the cluster is formed correctly and check the topology is complete.